ddtrace/contrib/asyncio/helpers.py

```diff
@@ -72,3 +73,54 @@ def _wrap_executor(fn, args, tracer, ctx):
     # fn() will be executed outside the asyncio loop as a synchronous code
     tracer._context_provider._local.set(ctx)
     return fn(*args)
+
+
+_orig_create_task = None
+
+
+def enable_task_linking():
```

Comment:
That sounds legit. As a first step, I'd suggest to split the `_create_task` so that it's a helper that can be used alone or with a hypothetical `enable_task_linking` function. Then, let's think about providing this functionality out of the box. We can convert the `enable_task_linking` function so that it's similar to what we usually do for other integrations:
* Tornado [example](https://github.com/thehesiod/dd-trace-py/blob/b9a41ec6cd7f8636cdbddb2134eaab88604f085e/ddtrace/contrib/tornado/patch.py#L11-L56)
* Gevent [example](https://github.com/thehesiod/dd-trace-py/blob/b9a41ec6cd7f8636cdbddb2134eaab88604f085e/ddtrace/contrib/gevent/patch.py#L12-L32)

So having a generic `patch()` function that is not called automatically. However, when it is, it could force the propagation between different Tasks. Also I think it could be great having the `patch()` and `unpatch()` idempotent like https://github.com/thehesiod/dd-trace-py/blob/b9a41ec6cd7f8636cdbddb2134eaab88604f085e/ddtrace/contrib/tornado/patch.py#L16-L19. So we don't cause an assert error if called twice.

In that way we can also add even more tests:
* using the helper to propagate the `Context` manually
* without the helper the propagation must not take place
* after calling the `patch()` the propagation must take place automatically
* after calling the `patch()` and then `unpatch()` the propagation must not take place

I think that these changes are reasonable so we have:
* a helper to manually propagate the `Context`
* a `patch()` method that patches asyncio (so developers are aware of that)
* the `patch()` is not called by default (so no huge / breaking changes after this merge)

What do you think? If it requires a lot of time, I may help with something.
Thank you!